test/adapter/json/nested_relationships_test.rb

```diff
@@ -0,0 +1,248 @@
+require 'test_helper'
+
+NestedPostSerializer = Class.new(ActiveModel::Serializer) do
+  attributes :id, :title
+
+  has_many :comments, include: :author
+end
+
+NestedCommentBelongsToSerializer = Class.new(ActiveModel::Serializer) do
+  attributes :id, :body
+
+  belongs_to :author, include: [:posts]
+end
+
+NestedAuthorSerializer = Class.new(ActiveModel::Serializer) do
+  attributes :id, :name
+
+  has_many :posts, include: [:comments]
+end
+
+ComplexNestedAuthorSerializer = Class.new(ActiveModel::Serializer) do
+  attributes :id, :name
+
+  # it would normally be silly to have this in production code, cause a post's
+  # author in this case is always going to be your root object
+  has_many :posts, include: [:author, comments: [:author]]
+end
+
+MultipleRelationshipAuthorSerializer = Class.new(ActiveModel::Serializer) do
+  attributes :id, :name
+
+  has_many :posts, include: [{ comments: [:author] }]
+  has_many :comments
+end
```

Comment:
If these are used only in this test, they should be defined inside `NestedRelationshipsTest` (not TestTest ) and not pollute the global scope